releasenotes/notes/deprecate-qasm-legacy-d0b6025493da453a.yaml

```diff
@@ -0,0 +1,29 @@
+---
+deprecations:
+  - |
+    The legacy OpenQASM 2 parser module previously present in :mod:`qiskit.qasm` has been
+    deprecated. It will be removed in the Qiskit 1.0.0 release. The legacy
+    OpenQASM 2 parser has been superseded by the :mod:`qiskit.qasm2` module
+    which provides a faster more correct parser for OpenQASM 2.
+  - |
+    The ``qiskit.converters.ast_to_dag`` function has been deprecated and
+    will be removed in the Qiskit 1.0.0 release. It previously was used to
+    convert the abstract syntax tree generated by the legacy OpenQASM 2 parser
+    (in the :mod:`qiskit.qasm` module which has been deprecated) and convert
+    that directly to a :class:`.DAGCircuit`. As the legacy OpenQASM 2 parser
+    has been deprecated this function will no longer serves a purpose after the
+    legacy parser is removed. If you were previously using this, you can
+    instead parse your OpenQASM 2 files into a :class:`.QuantumCircuit` using
+    the :meth:`.QuantumCircuit.from_qasm_file` or
+    :meth:`.QuantumCircuit.from_qasm_str` constructor methods and then
```

Comment:
I may have forgotten about this in the other PR, but weren't we planning to get rid of these and require use of `qasm2.loads` as well?

Reply:
We kept them in the other PR. At least for me the circuit constructor class methods still have value just in that I don't need to remember all the extra arguments to the parser to get it to behave like the legacy parser did (which is the only mode I ever really use for parsing qasm2).

Reply:
That's fair.  I think we ought to remove those methods once we've implemented #10737 and made `qasm2.dumps` match it, at which point `qasm2.loads` will load the output of `qasm2.dumps` without any additional modifications. The arguments in favour of removing `QuantumCircuit.qasm` also imo hold for `from_qasm_str`, it's just the usability concern that you're correctly pointing out in the way, I think.